### PR TITLE
Feature/prerender image support

### PR DESCRIPTION
**Pull request by dac09**
Builds on top of #1641

### What?
When rendering in the Node.js environment, babel doesn't know how to handle images. 

This PR does the following things:
- handle SVGs using a babel plugin, instead of webpack's loader
- adds webpack manifest plugin, which generates a manifest.json with all the files that have been processed by webpack
- adds new babel plugin for prerender, which looks at the manifest.json, and replaces any media files (imgs, fonts,pdfs,etc.) with the path generated by webpack
- for any files that aren't in the manifest (which means its handled by the url loader), it puts in place a blank transparent gif data url. There's room for improvement here, for sure! But instead of putting the effort here, we should think about how to optimise the loading of images similar to next-images

### Why?
So that we can prerender pages where media is imported like
```
import cuteKittenSrc from './cuteKitten.png'
```

--- 
### Other notes

**1. I've noticed there are subtle differences in how SVGs are handled. For example, this won't work anymore:**

Logo.js
```
import Logo from './Logo.svg'
export default Logo
```

but this will

```
import Logo from './Logo.svg'
const LogoImage = () => <Logo/>

export default LogoImage

```

### CODE REVIEW

**Comment by dac09** [comment]
Converting back to draft, I will rebase the relevant code after merge of #1641

**Comment by dac09** [comment]
@peterp ready for review.

File: packages/core/config/webpack.common.js [hunk 1]
```diff
@@ -250,12 +250,7 @@ module.exports = (webpackEnv) => {
                 },
               },
             },
-            // (2)
-            {
-              test: /\.svg$/,
-              loader: 'svg-react-loader',
```
Comment (peterp) [inline packages/core/config/webpack.common.js:256]: This package should be removed from `core/package.json`

File: packages/prerender/package.json [hunk 2]
```diff
@@ -22,19 +21,16 @@
     "@redwoodjs/router": "^0.25.1",
     "@redwoodjs/web": "^0.25.1",
     "babel-plugin-ignore-html-and-css-imports": "^0.1.0",
-    "babel-plugin-inline-react-svg": "^1.1.2",
-    "node-fetch": "^2.6.1"
-  },
-  "devDependencies": {
-    "nodemon": "^2.0.6"
+    "node-fetch": "^2.6.1",
+    "nodemon": "^2.0.7"
```
Comment (peterp) [inline packages/prerender/package.json:25]: Is nodemon not a dev-dep?

File: packages/prerender/package.json [hunk 3]
```diff
@@ -6,13 +6,12 @@
   "files": [
     "dist",
     "dist/index.js",
```
Comment (peterp) [inline packages/prerender/package.json:8]: It should be fine to just specify "dist" as a top level, but we should also include "browserUtils"
```suggestion
    "browserUtils"
```

File: packages/prerender/package.json [hunk 4]
```diff
@@ -6,13 +6,12 @@
...
-		"dist/browerUtils/index.js"
+    "dist/browerUtils/index.js"
```
Comment (peterp) [inline packages/prerender/package.json:9]:
```suggestion
```

**Review by peterp — CHANGES_REQUESTED** [review]
This looks good to me, just a few changes to the package.json files.

I'm a little concerned about the having to wrap the SVG imports into functions, since that would introduce a breaking change? As an example:

```js
import MyIcon from './MyIcon.svg'

export const MyComponent = () => {
   <div>
       <MyIcon />
   </div>
}
```

Would this not work anymore? It's odd to me, since the documentation seems to indicate that it creates a component.

If this is the case then this is a huge breaking change.

**Comment by thedavidprice** [comment]
@dac09 I took this for a spin locally. (Note: I'm having a lot of trouble these days with my local dev setup. PR Packages don't seem to be created correctly, i.e. the inter-dependencies aren't actually to the PR package versions. Verdaccio was failing. And `rwt cpw` is hit or miss. No idea...)

Got this code to prerender:
```js
// .../BlogLayout.js
...

import FaviconImage from './favicon.png'

const Favicon = () => <FaviconImage />

const BlogLayout = ({ children }) => {
  const { logIn, logOut, isAuthenticated, currentUser } = useAuth()

  return (
    <>
      <header>
        <h1>
          <img src={Favicon} /> <Link to={routes.home()}>Redwood Blog</Link>
        </h1>
...
```

Ran `rw build` successfully.

But when I ran `rw prerender -d`, I am getting this:
```bash
Could not resolve "/Users/price/Repos/tdp-tutorial-blog-test/web/src/pages/FatalErrorPage" in file /Users/price/Repos/tdp-tutorial-blog-test/web/src/index.js.
Could not resolve "/Users/price/Repos/tdp-tutorial-blog-test/web/src/layouts/BlogLayout" in file /Users/price/Repos/tdp-tutorial-blog-test/web/src/pages/AboutPage/AboutPage.js.
Warning: Invalid value for prop `src` on <img> tag. Either remove it from the element, or pass a string or number value to keep it in the DOM. For details, see https://fb.me/react-attribute-behavior
...
::: Dry run, not writing changes :::
::: 🚀 Prerender output for /about ::: 
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width,initial-scale=1" />
    <link rel="icon" type="image/png" href="/favicon.png" />
    <title>tdp-tutorial-blog-test</title>
    <script defer="defer" src="/static/js/runtime-app.31ca6fe6.js"></script>
    <script defer="defer" src="/static/js/vendors.9e95d691.chunk.js"></script>
    <script defer="defer" src="/static/js/app.87f08f92.chunk.js"></script>
    <link href="/static/css/app.6d4df765.css" rel="stylesheet" />
  </head>
  <body>
    <div id="redwood-app">
      <header>
        <h1><img /> <a href="/">Redwood Blog</a></h1>
...
```

**Comment by dac09** [comment]
> This looks good to me, just a few changes to the package.json files.
> 
> 
> 
> I'm a little concerned about the having to wrap the SVG imports into functions, since that would introduce a breaking change? As an example:
> 
> 
> 
> ```js
> 
> import MyIcon from './MyIcon.svg'
> 
> 
> 
> export const MyComponent = () => {
> 
>    <div>
> 
>        <MyIcon />
> 
>    </div>
> 
> }
> 
> ```
> 
> 
> 
> Would this not work anymore? It's odd to me, since the documentation seems to indicate that it creates a component.
> 
> 
> 
> If this is the case then this is a huge breaking change.
> 
> 

@peterp that would definitely work. It's only when you are "re-exporting" an image that it doesn't work.

File: packages/prerender/package.json (hunk 2 above)
Comment (dac09) [reply packages/prerender/package.json:25]: Whoops, we don't even need it here

File: packages/core/config/webpack.common.js (hunk 1 above)
Comment (dac09) [reply packages/core/config/webpack.common.js:256]: Done

**Review by peterp — APPROVED** [review]

**Comment by dac09** [comment]
> Got this code to prerender:
> 
> ```js
> // .../BlogLayout.js
> ...
> 
> import FaviconImage from './favicon.png'
> 
> const Favicon = () => <FaviconImage />
> 
> const BlogLayout = ({ children }) => {
>   const { logIn, logOut, isAuthenticated, currentUser } = useAuth()
> 
>   return (
>     <>
>       <header>
>         <h1>
>            <Link to={routes.home()}>Redwood Blog</Link>
>         </h1>
> ...
> ```
> 
> Ran `rw build` successfully.
> 
> But when I ran `rw prerender -d`, I am getting this:
> 

There's a syntax error there, the way imported images work (unless they're svg) is:
```
import FaviconImage from './favicon.png'

return (
    <>
      <header>
        <h1>
          <img src={FaviconImage} /> <Link to={routes.home()}>Redwood Blog</Link>
        </h1>
 ...
```

So you can skip that extra `Favicon` component :)

**Comment by thedavidprice** [comment]
Ah, got it. OK, gonna need more. Doc. POWER 🦾⚡️

**Comment by dac09** [comment]
I think its documented somewhere, definitely seen it before. Defo worth exposing a bit more though

 its maintaining the behaviour from before, not introducing something new!

**Pull request merged**